bycycle/tests/cyclepoints/test_extrema.py

```diff
@@ -37,12 +37,12 @@ def test_find_extrema(first_extrema):
     minima = argrelextrema(sig, np.less)
 
     # Find peaks and troughs using bycycle and make sure match scipy
-    peaks, troughs = find_extrema(sig, fs, f_range, boundary=1, first_extrema=first_extrema)
+    peaks, troughs = find_extrema(sig, fs, f_range, first_extrema=first_extrema)
 
     if first_extrema == 'trough':
         assert len(peaks) == len(troughs)
         assert troughs[0] < peaks[0]
-        np.testing.assert_allclose(peaks, maxima[0])
+        np.testing.assert_allclose(peaks, maxima[0][:len(peaks)])
```

Comment:
It's because when first_extrema is trough, the last extrema must be a peak and vice-versa. This will cause beginning/ending extrema to sometimes cut off peaks/troughs (i.e. first extrema is actually a peak, but `first_extrema` is set to `trough`, then the first actual peak (found with `argrelextrema`) will be missing, etc).

I've updated with comments explaining this.